pysmt/solvers/z3.py

```diff
@@ -147,14 +147,14 @@ def __init__(self, environment, logic, **options):
                                            environment=environment,
                                            logic=logic,
                                            **options)
-        try:
-            self.z3 = z3.SolverFor(str(logic))
-        except z3.Z3Exception:
-            self.z3 = z3.Solver()
-        except z3.z3types.Z3Exception:
-            self.z3 = z3.Solver()
-        except OSError:
-            self.z3 = z3.Solver()
+        # try:
+        #     self.z3 = z3.SolverFor(str(logic))
+        # except z3.Z3Exception:
+        #     self.z3 = z3.Solver()
+        # except z3.z3types.Z3Exception:
+        #     self.z3 = z3.Solver()
+        # except OSError:
+        self.z3 = z3.Solver()
```

Comment:
This does not seem related to the pytest transition. Was this related to how we resolve the import of z3?

Reply:
I think this was the source of the damn segfault on the CI... But you are right, it is unrelated. Shall we move to a separate PR or keep it?

Reply:
Let's move it out and make sure this is indeed the reason of the CI segfault. I also do not remember why that code might have been there in the first place, so a more thorough review would be prefereable.

Reply:
OK